substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -3263,31 +3349,40 @@ impl<T: Config> Pallet<T> {
 }
 
 impl<T: Config> sp_staking::OnStakingUpdate<T::AccountId, BalanceOf<T>> for Pallet<T> {
+	/// Reduces the balances of the [`SubPools::with_era`], that belong to the pool involved in the
+	/// slash, to the amount that is defined in the `slashed_unlocking` field of
+	/// [`sp_staking::OnStakingUpdate::on_slash`]
+	///
+	/// Emits [`Event::<T>::PoolsSlashed`].
 	fn on_slash(
 		pool_account: &T::AccountId,
 		// Bonded balance is always read directly from staking, therefore we don't need to update
 		// anything here.
 		slashed_bonded: BalanceOf<T>,
 		slashed_unlocking: &BTreeMap<EraIndex, BalanceOf<T>>,
+		total_slashed: BalanceOf<T>,
 	) {
-		if let Some(pool_id) = ReversePoolIdLookup::<T>::get(pool_account) {
-			let mut sub_pools = match SubPoolsStorage::<T>::get(pool_id).defensive() {
-				Some(sub_pools) => sub_pools,
-				None => return,
-			};
-			for (era, slashed_balance) in slashed_unlocking.iter() {
-				if let Some(pool) = sub_pools.with_era.get_mut(era) {
+		let Some(pool_id) = ReversePoolIdLookup::<T>::get(pool_account) else { return };
+		// As the slashed account belongs to a `BondedPool` the `TotalValueLocked` decreases and
+		// an event is emitted.
+		TotalValueLocked::<T>::mutate(|tvl| {
+			tvl.saturating_reduce(total_slashed);
```

Comment:
Should be `checked_sub` and a defensive assertion since slashed should never be higher than bonded amount (and hence never higher than TVL).

Reply:
So currently any slash higher than `TVL` would result in `TVL` becoming zero but without any defensive error being provided notifying about the unreasonably high slash. 

As you now wish for a `checked_sub` what would be the expected outcome? Should it still be 0? Do you want to retain the original `TVL` and declare this slash as invalid? Whats the goal of this change? Just a defensive error in case the `total_slashed` would be larger than `TVL`?

Reply:
Assuming that it's only about the defensive error I switched to `defensive_saturating_reduce`